app/models/stack.rb

```diff
@@ -244,6 +244,14 @@ def enable_ci!
     Rails.cache.write(ci_enabled_cache_key, true)
   end
 
+  def mark_as_inaccessible!
+    update!(inaccessible_since: Time.now) unless inaccessible?
+  end
+
+  def inaccessible?
+    inaccessible_since?
```

Comment:
Isn't `inaccessible_since` a datetime? Does Rails generate `?` methods for all attributes?